```
+=======================================
|  Tests summary:
|---------------------------------------
| test_flake8                 FAIL
| test_pylint                 Success
+=======================================
```

Por lo que los errores son de PEP8
```
account_invoice_automatic_pay_email/hooks.py:5:1: F401 'openerp.api' imported but unused
account_invoice_automatic_pay_email/hooks.py:5:1: F401 'openerp.SUPERUSER_ID' imported but unused
```

--- a/account_invoice_automatic_pay_email/README.rst
+++ b/account_invoice_automatic_pay_email/README.rst
@@ -1,0 +1,18 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+    :alt: License: AGPL-3
+
+===================================
+Account invoice automatic pay email
+===================================
+
+This module creates new field 'payment reminder date' in account.invoice to
+send payment reminder emails to customers every seven days when reminder date
+has expired.
+
+Credits
+=======
+
+Contributors
+------------
+* Gotzon Imaz <gotzonimaz@avanzosc.es>

--- a/account_invoice_automatic_pay_email/__init__.py
+++ b/account_invoice_automatic_pay_email/__init__.py
@@ -1,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Gotzon Imaz - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from . import models

--- a/account_invoice_automatic_pay_email/__openerp__.py
+++ b/account_invoice_automatic_pay_email/__openerp__.py
@@ -1,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Gotzon Imaz - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+{
+    "name": "Account Invoice Automatic Pay Email",
+    'version': '8.0.1.0.0',
+    'license': "AGPL-3",
+    'author': "AvanzOSC",
+    'website': "http://www.avanzosc.es",
+    'contributors': [
+        "Gotzon Imaz <gotzonimaz@avanzosc.es",
+    ],
+    "category": "Pay email",
+    "depends": [
+        'account',
+    ],
+    "data": [
+        'data/account_invoice_automatic_pay_email_data.xml',
+        'views/account_invoice_view.xml'
+    ],
+    "installable": True,
+}

--- a/account_invoice_automatic_pay_email/data/account_invoice_automatic_pay_email_data.xml
+++ b/account_invoice_automatic_pay_email/data/account_invoice_automatic_pay_email_data.xml
@@ -1,0 +1,31 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data noupdate="1">
+        <record forcecreate="True" id="ir_cron_automatic_payment_reminder_action" model="ir.cron">
+            <field name="name">Automatic payment reminder</field>
+            <field eval="False" name="active" />
+            <field name="user_id" ref="base.user_root" />
+            <field name="interval_number">1</field>
+            <field name="interval_type">days</field>
+            <field name="numbercall">-1</field>
+            <field eval="'account.invoice'" name="model" />
+            <field eval="'automatic_pay_email'" name="function" />
+            <field eval="'(None, )'" name="args" />
+        </record>
+        <record id="email_customers_payment_reminder" model="email.template">
+            <field name="name">Email to customers payment reminder</field>
+            <field name="model_id" ref="account.model_account_invoice"/>
+            <field name="auto_delete" eval="True"/>
+            <field name="reply_to">${(object.company_id.email or 'noreply@localhost')|safe}</field>
+            <field name="subject">Invoice ${object.number} payment pending.</field>
+            <field name="lang">${object.partner_id.lang}</field>
+            <field name="body_html"><![CDATA[
+    <p>Dear customer:</p>
+    <p>We contact you in regarding invoice number ${object.number} dated ${object.date_invoice}, in the amount of ${object.amount_total} euros, mature on the ${object.date_due}.</p>
+    <p>The subject of this notification is that to date we haven't received the payment confirmation of the mencioned invoice, therefore we would ask you to make the payment as soon as possible.</p>
+    <p>If you have any doubt don't hesitate to contact me.</p>
+    <p>That is all for the present,  I would like to take this opportunity to extend a cordial greeting.</p>
+            ]]></field>
+        </record>
+    </data>
+</openerp>

--- a/account_invoice_automatic_pay_email/hooks.py
+++ b/account_invoice_automatic_pay_email/hooks.py
@@ -1,0 +1,11 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Gotzon Imaz - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+
+def post_init_hook(cr, pool):
+    cr.execute(
+        """
+        UPDATE account_invoice
+        SET payment_reminder_date = date_due;
+        """)

--- a/account_invoice_automatic_pay_email/i18n/es.po
+++ b/account_invoice_automatic_pay_email/i18n/es.po
@@ -1,0 +1,47 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+#	* account_invoice_automatic_pay_email
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 8.0\n"
+"Report-Msgid-Bugs-To: \n"
+"POT-Creation-Date: 2018-07-26 10:13+0000\n"
+"PO-Revision-Date: 2018-07-26 10:13+0000\n"
+"Last-Translator: <>\n"
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: account_invoice_automatic_pay_email
+#: model:ir.model,name:account_invoice_automatic_pay_email.model_account_invoice
+msgid "Invoice"
+msgstr "Factura"
+
+#. module: account_invoice_automatic_pay_email
+#: field:account.invoice,payment_reminder_date:0
+msgid "Payment reminder date"
+msgstr "Fecha de aviso"
+
+#. module: account_invoice_automatic_pay_email
+#: model:email.template,body_html:account_invoice_automatic_pay_email.email_customers_payment_reminder
+msgid "\n"
+"    <p>Dear customer:</p>\n"
+"    <p>We contact you regarding invoice number ${object.number} dated ${object.date_invoice}, in the amount of ${object.amount_total} euros, mature on the ${object.date_due}.</p>\n"
+"    <p>The subject of this notification is that to date we haven't received the payment confirmation of the mentioned invoice, therefore we would ask you to make the payment as soon as possible.</p>\n"
+"    <p>If you have any doubt don't hesitate to contact me.</p>\n"
+"    <p>That is all for the present,  I would like to take this opportunity to extend a cordial greeting.</p>\n"
+"            "
+msgstr "\n"
+"    <p>Estimado cliente:</p>\n"
+"    <p>Nos ponemos en contacto con usted en referencia a la factura número ${object.number} de fecha ${object.date_invoice}, por importe de ${object.amount_total} euros, con vencimiento el día ${object.date_due}.</p>\n"
+"    <p>El motivo de la comunicación es que hasta la fecha no hemos recibido la confirmación del pago de la factura mencionada, por lo que le rogaríamos que efectuara el abono lo antes posible.</p>\n"
+"    <p>Si tiene cualquier duda al respecto puede ponerse en contacto conmigo.</p>\n"
+"    <p>Sin otro particular, aprovecho para enviarle un cordial saludo.</p>\n"
+"            "
+#. module: account_invoice_automatic_pay_email
+#: model:email.template,subject:account_invoice_automatic_pay_email.email_customers_payment_reminder
+msgid "Invoice ${object.number} payment pending."
+msgstr "Factura ${object.number} pago pendiente."

--- a/account_invoice_automatic_pay_email/models/__init__.py
+++ b/account_invoice_automatic_pay_email/models/__init__.py
@@ -1,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Gotzon Imaz - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from . import account_invoice

--- a/account_invoice_automatic_pay_email/models/account_invoice.py
+++ b/account_invoice_automatic_pay_email/models/account_invoice.py
@@ -1,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Gotzon Imaz - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, api, fields
+from dateutil.relativedelta import relativedelta
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    payment_reminder_date = fields.Date(
+        string='Payment reminder date')
+
+    @api.multi
+    def automatic_pay_email(self):
+        mail_template = self.env.ref(
+            'account_invoice_automatic_pay_email.email_customers_payment_'
+            'reminder', False)
+        invoice_obj = self.env['account.invoice']
+        if mail_template:
+            cond = [('payment_reminder_date', '<=',
+                     fields.Date.context_today(self)),
+                    ('state', '=', 'open'),
+                    ('type', '=', 'out_invoice'),
+                    ('payment_term', '!=', False)]
+            invoices = invoice_obj.search(cond)
+            for invoice in invoices:
+                if (invoice.payment_term.line_ids and
+                        invoice.payment_term.line_ids[0].days > 0):
+                    mail = self.env['mail.compose.message'].with_context(
+                        default_composition_mode='mass_mail',
+                        default_template_id=mail_template.id,
+                        default_use_template=True,
+                        default_partner_ids=[(6, 0, invoice.partner_id.ids)],
+                        active_id=self.id,
+                        active_ids=self.ids,
+                        active_model='account.invoice',
+                        default_model='account.invoice',
+                        default_res_id=invoice.id,
+                        force_send=True
+                    ).create({'subject': mail_template.subject,
+                              'body': mail_template.body_html})
+                    mail.send_mail()
+                    today = fields.Date.context_today(self)
+                    invoice.payment_reminder_date = fields.Date.to_string(
+                        fields.Date.from_string(today) +
+                        relativedelta(days=7))
+
+    @api.multi
+    def onchange_payment_term_date_invoice(self, payment_term_id,
+                                           date_invoice):
+        result = super(
+            AccountInvoice, self).onchange_payment_term_date_invoice(
+            payment_term_id, date_invoice)
+        if result.get('value', False).get('date_due', False):
+            result['value']['payment_reminder_date'] = (
+                result.get('value').get('date_due'))
+        return result

--- a/account_invoice_automatic_pay_email/tests/__init__.py
+++ b/account_invoice_automatic_pay_email/tests/__init__.py
@@ -1,0 +1,4 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Gotzon Imaz - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from . import test_account_invoice_automatic_pay_email

--- a/account_invoice_automatic_pay_email/tests/test_account_invoice_automatic_pay_email.py
+++ b/account_invoice_automatic_pay_email/tests/test_account_invoice_automatic_pay_email.py
@@ -1,0 +1,48 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+import openerp.tests.common as common
+from openerp import fields
+from dateutil.relativedelta import relativedelta
+
+
+class TestAccountInvoiceAutomaticPayEmail(common.TransactionCase):
+
+    def setUp(self):
+        super(TestAccountInvoiceAutomaticPayEmail, self).setUp()
+        self.invoice_model = self.env['account.invoice']
+        self.today = fields.Date.from_string(fields.Date.today())
+        self.partner = self.browse_ref('base.res_partner_1')
+        self.partner.email = 'automaticpayemail@odoo.com'
+        self.payment_term = self.browse_ref(
+            'account.account_payment_term_15days')
+        invoice_vals = {
+            'partner_id': self.partner.id,
+            'account_id': self.browse_ref('account.a_pay').id,
+            'type': 'out_invoice',
+            'payment_term': self.payment_term.id}
+        product = self.browse_ref('product.product_product_6')
+        account = self.browse_ref('account.analytic_online')
+        line_vals = {
+            'product_id': product.id,
+            'name': product.name,
+            'account_analytic_id': account.id,
+            'quantity': 1,
+            'price_unit': 800}
+        invoice_vals['invoice_line'] = [(0, 0, line_vals)]
+        self.invoice = self.invoice_model.create(invoice_vals)
+        self.invoice.signal_workflow('invoice_open')
+
+    def test_account_invoice_auotmatic_pay_email(self):
+        res = self.invoice.onchange_payment_term_date_invoice(
+            self.payment_term.id, '2017-01-01')
+        value = res.get('value')
+        self.assertEqual(
+            value.get('payment_reminder_date'), '2017-01-16',
+            'Bad payment reminder date(1)')
+        self.invoice.payment_reminder_date = '2017-01-16'
+        self.invoice_model.automatic_pay_email()
+        self.assertEqual(
+            fields.Date.from_string(self.invoice.payment_reminder_date),
+            self.today + relativedelta(days=7),
+            'Bad payment reminder date(2)')

--- a/account_invoice_automatic_pay_email/views/account_invoice_view.xml
+++ b/account_invoice_automatic_pay_email/views/account_invoice_view.xml
@@ -1,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<openerp>
+    <data>
+        <record model="ir.ui.view" id="view_account_invoice_inh_form">
+            <field name="name">view.account.invoice.inh.form</field>
+            <field name="model">account.invoice</field>
+            <field name="inherit_id" ref="account.invoice_form" />
+            <field name="arch" type="xml">
+                <field name="date_due" position="after">
+                        <field name="payment_reminder_date" />
+                </field>
+            </field>
+        </record>
+         <record id="account_invoice_automatic_email_view_tree" model="ir.ui.view">
+                <field name="name">account.invoice.automatic.email.view.tree</field>
+                <field name="model">account.invoice</field>
+                <field name="inherit_id" ref="account.invoice_tree" />
+                <field name="arch" type="xml">
+                    <field name="date_due" position="after">
+                            <field name="payment_reminder_date"/>
+                    </field>
+                </field>
+         </record>
+    </data>
+</openerp>

